R/error_checking.R

```diff
@@ -0,0 +1,340 @@
+{
+    # Future
+    # 	Does it have nulls?
+    #  	Warning that pops up to the user about null values
+
+    result <- "Success"
+
+    # Verify map_data
+    if(!is.null(map_data))
+    {
+        # Verify data_key_field
+        if(!is.null(data_key_field))
+        {
+            if(!"character" %in% class(data_key_field))
+                return("Error: data_key_field argument must be of type character")
+            if(!data_key_field %in% colnames(map_data))
+                return("Error: data_key_field was not found in the map_data data frame")
+        }
+
+        # Verify data_col
+        if(!is.null(data_col))
+        {
+            if(!"character" %in% class(data_col))
+                return("Error: data_col argument must be of type character")
+            if(!data_col %in% colnames(map_data))
+                return("Error: data_col was not found in the map_data data frame")
+        }
+    }
+
+    return(result)
+}
+
+
+#' Verify csv data
+#'
+#' This function runs a check on csv data inputs and looks for errors
+#' @param map_data (Data Frame or Character) - Either the full path string to data file (csv) or a data.frame object
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_csv <- function(map_data)
+{
+
+    # Check csv for things like character Unicode
+    # Check field they provided there etc
+
+
+    result <- "Success"
+
+    return(result)
+
+}
+
+
+#' Verify map parameters
+#'
+#' @param bin_method (Character) - Method or function to use to split continuous data into discrete chunks (one of "quantile", "equal", "pretty", "kmeans") (default "pretty")
+#' @param bins (Numeric) - Number of bins/segments in which to divide the raster
+#' @param dpi (Numeric) - Settable DPI for different print/screen formats (default 150)
+#' @param expand_xy (c(Numeric, Numeric)) - Sets expansion amount for the X and Y scale of the map - Vector (expand x, expand y) (default c(0,0))
+#' @param map_xy_min_max (c(Numeric, ...)) - Vector that describes the desired extent of the map in the form of (xmin, xmax, ymin, ymax) (default: c(-180, 180, -90, 90))
+#' @param map_title (Character) - Title to be displayed on the output map
+#' @param map_palette (Character) - Optional variable to manually set the colorscale to a specific palette from RColorbrewer
+#' @param map_palette_reverse (Boolean) - Set palette to reverse direction TRUE/FALSE
+#' @param map_width_height_in (c(Numeric, Numeric)) - Vector that describes the desired file size of the output image in the form of (width, height) in inches (default c(15, 10))
+#' @param map_legend_title (Character) - Text for the legend header
+#' @param map_x_label (Character) - Label for x axis (default Lon)
+#' @param map_y_label (Character) - Label for y axis (default Lat)
+verify_map_params <- function(bin_method, bins, dpi, expand_xy, map_xy_min_max , map_title, map_palette, map_palette_reverse,
```

Comment:
No export for this one?  Just checking, may be fine

Reply:
I wasn't sure how things are done here, if exports are selectively picked or just mostly exported unless there's a particular reason for it. I see the benefits of both so let me know and I can export most things. In this case, the only thinking behind no export was simply I that I couldn't think of a reason why someone would want to call this function externally.